tests/functional/amo_sign_up.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern',
+  'intern!object',
+  'require',
+  'tests/functional/lib/helpers',
+  'tests/functional/lib/fx-desktop'
+], function (intern, registerSuite, require, FunctionalHelpers, FxDesktopHelpers) {
+  var config = intern.config;
+  var PAGE_URL = config.fxaContentRoot + 'signin?context=fx_desktop_v1';
+  var PAGE_URL_WITH_MIGRATION = PAGE_URL + '&migration=amo';
+
+  var SIGNIN_URL = config.fxaContentRoot + 'signin';
+
+  var listenForFxaCommands = FxDesktopHelpers.listenForFxaCommands;
+
+  registerSuite({
+    name: 'Firefox AMO sign_up',
+
+    beforeEach: function () {
+      return FunctionalHelpers.clearBrowserState(this);
+    },
+
+    afterEach: function () {
+      var self = this;
+
+      return FunctionalHelpers.clearBrowserState(self)
+        .then(function () {
+          // ensure the next test suite (bounced_email) loads a fresh
```

Comment:
Looking at where the tests are inserted in the list above, I'm not convinced this comment is true. Can you check to ensure this `.then` is necessary?